### DISCUSSION TITLE
求助，如何获取动态对象的实时sampledPosition值？

### DISCUSSION THREAD

**Discussion by KKindom** [zouyaoji/vue-cesium#252] [Q&A] [answered]
请问下，我想获取小车的每秒的sampledPosition，使用了@stop-arrived="stopArrived"但是如果设置时间倍速之后他就更新的很慢，我这边设置了一个时间监听，想获取小车的位置信息用来显示不同的车速，如何解决？我看源码里面stopArrived里面使用的
 const currentPosition = overlay._sampledPosition.getValue(clock.currentTime)，我也想用，但是报错说sampledPosition没有getValue方法，请问下怎么实现？

**Comment by zouyaoji** [comment, marked as answer]
参考一下在线文档例子哈。获取速度不是sampledPosition吧。

https://zouyaoji.top/vue-cesium/#/zh-CN/component/overlays/vc-overlay-dynamic
标签那里有一句
const velocityVector = dynamicOverlayRef.value.getOverlays()[0]._velocityVectorProperty.getValue(time, {})
可以用这个获取速度。

另外我这 overlay._sampledPosition是有getValue方法的。你看看你vue-cesium是不是最新版本哈？

**Reply by KKindom** [reply]
感谢回复，发现我老师用的版本不是新的，我这里需求可能比较奇怪现在用平均速度😄，后面可能才会改成实时的速度。我现在解决了，再试试作者的方法看看。
我是用模拟时间戳和每段路到达时间的时间戳做对比，然后设置text的文本内容。

**Reply by zouyaoji** [reply]
嗯，感觉应该在stop-arrived事件把时间也传出来，我下个版本加一下。

**Reply by KKindom** [reply]
对这里有一个问题，我不知道是我的设置还是什么，就是把时间倍速调高的时候，stop-arrived的回调会变得很奇怪，不是每几秒调用一次，有时候一次性调用一堆，然后10多秒没反应，这是啥情况，我之前用stop-arrived监听就是这个问题搞得我那个速度一直没法正常变化

**Reply by zouyaoji** [reply]
multiplier 调高了，时钟也是突变的，很容易导致组件判断是否到达站点的方法不准。
https://github.com/zouyaoji/vue-cesium/blob/dev/packages/components/overlays/dynamic/index.ts#L370
目前也没想到好的办法，可以尝试不改multiplier ，而是把站点之间的间隔时间变短。
